packages/common/src/common.ts

```diff
@@ -779,6 +780,26 @@ export class Common {
     return null
   }
 
+  /**
+   * Returns the hardfork change block for eip
+   * @param eip EIP number
+   * @returns Block timestamp or null if unscheduled
+   */
+  eipTimestamp(eip: number): bigint | null {
+    // NOTE: this is not elegant, what if the eip got activated on a block and not timestamp?
```

Comment:
Don't the older hardfork block headers still have a timestamp in them that we can report here?  I'm not sure if this is really a concern.  Or, maybe better yet, we just return `null` for any EIP that's not scheduled or pre-Shanghai.

Reply:
Oh right, this is indeed an interesting line of thought. However this would make the method `async` and we would need to get the block from `blockchain` in order to retrieve the timestamp :thinking: Not sure I like that. Especially gets a bit hairy if it reorgs (you save the timestamp, chain reorgs, and now the timestamp differs). I will update the docs to remove the "block" part and just write down that it returns the timestamp when this EIP is scheduled (therefore, on missed slots the first block might actually be later than the scheduled timestamp)

Reply:
Ahh! Wait! I now remember the point I wanted to make here. The same applies to `eipBlock` actually. If `eipTimestamp` returns `null` (for instance, for `Dao`), then we know it is either (1) not scheduled at a timestamp (but at a block number) or (2) it is not even scheduled at all (so the fork will never happen). I think this is beyond the scope of this PR though and it is also an edge case.